src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -110,17 +138,39 @@ const useDiscovery = () => {
       setRefreshing(false);
     }
   };
+
+  const getIsMeFollowingTargetStatus = React.useCallback(
+    (userId) => {
+      const targetUser = discovery?.initialUsers?.find((item) => item?.user_id === userId);
+      const targetUserFromFollowed = discovery?.followedUsers?.find(
+        (item) => item?.user_id === userId
+      );
+      const targetUserFromFollowing = discovery?.unfollowedUsers?.find(
+        (item) => item?.user_id === userId
+      );
+
+      return (
+        targetUser?.is_followed ||
+        targetUserFromFollowed?.is_followed ||
+        targetUserFromFollowing?.is_followed ||
+        false
+      );
+    },
+    [discovery]
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of the `getIsMeFollowingTargetStatus` function improves the modularity of the code by encapsulating the logic for checking if the current user follows a target user. This makes the code easier to understand and maintain.



<!-- This is an auto-generated comment by OSS CodeRabbit -->